src/components/Card.js

```diff
@@ -1,32 +1,63 @@
+// import { api } from "../components/Api.js";
+
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLike,
+    handleDislike
+  ) {
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._cardElement = null;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._id = _id;
+    this._isLiked = isLiked;
+    this._handleLike = handleLike;
+    this._handleDislike = handleDislike;
+    this._handleLikeButtonClick = this._handleLikeButtonClick.bind(this);
+  }
+
+  _handleLikeButtonClick() {
+    if (this._isLiked) {
+      this._handleDislike(this);
+    } else {
+      this._handleLike(this);
+    }
   }
 
+  _handleFormSubmit = (event) => {
+    event.preventDefault();
+    const popup = document.getElementById("trash_button");
+    popup.classList.remove("popup__active");
+    this._cardElement.remove();
```

Comment:
Please, delete it

You don't use it and you can't use it like that


And modal is handled by it's class